From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Eric Kilmer <eric.d.kilmer@gmail.com>
Date: Sat, 30 Mar 2024 18:26:27 -0400
Subject: [PATCH] 6372: decompiler-cpp: Open sla files as 'binary'

Compressed files need to be opened as binaries when reading. Fixes
opening sla files on Windows.
---
 Ghidra/Features/Decompiler/src/decompile/cpp/sleigh.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Ghidra/Features/Decompiler/src/decompile/cpp/sleigh.cc b/Ghidra/Features/Decompiler/src/decompile/cpp/sleigh.cc
index d26914761f..beb3406b0f 100644
--- a/Ghidra/Features/Decompiler/src/decompile/cpp/sleigh.cc
+++ b/Ghidra/Features/Decompiler/src/decompile/cpp/sleigh.cc
@@ -559,7 +559,7 @@ void Sleigh::initialize(DocumentStorage &store)
     if (el == (const Element *)0)
       throw LowlevelError("Could not find sleigh tag");
     sla::FormatDecode decoder(this);
-    ifstream s(el->getContent());
+    ifstream s(el->getContent(), std::ios_base::binary);
     if (!s)
       throw LowlevelError("Could not open .sla file: " + el->getContent());
     decoder.ingestStream(s);
-- 
2.44.0

